src/main/java/edu/kis/vh/nursery/HanoiRhymer.java

```diff
@@ -1,17 +1,29 @@
 package edu.kis.vh.nursery;
 
+/**
+ * searching rhymer using Hanoi sequence
+ */
 public class HanoiRhymer extends DefaultCountingOutRhymer {
 
-    int totalRejected = 0;
+    private int totalRejected = 0;
 
-    public int reportRejected() {
-        return totalRejected;
+    protected int reportRejected() {
+        return getTotalRejected();
     }
 
-    public void countIn(int in) {
+    @Override
+    protected void countIn(int in) {
         if (!callCheck() && in > peekaboo())
-            totalRejected++;
+            setTotalRejected(getTotalRejected() + 1);
         else
             super.countIn(in);
     }
+
+    public int getTotalRejected() {

```

Comment:
Shouldn't be encapsulated (typical private field). This is an exaggeration. Using private fields internally usually do not require encapsulation.